File: flightsBackend/src/main/java/com/example/flightsBackend/entities/flightData/FlightData.java
package com.example.flightsBackend.entities.flightData;

import lombok.Data;

@Data
public class FlightData {

    private String type;
    private Long id;
    private String source;
    private Boolean instantTicketingRequired;
    private Boolean nonHomogeneous;
    private Boolean oneWay;
    private String lastTicketingDate;
    private Integer numberOfBookableSeats;
    private Itineraries[] itineraries;
    private Price price;
}

File: flightsFrontend/src/app/model/flightData/flight-data.ts
import { Itineraries } from './itineraries';
import { Price } from './price';

export class FlightData {
    type: string;
    id: number;
    source: string;
    instantTicketingRequired: boolean;
    nonHomogeneous: boolean;
    oneWay: boolean;
    lastTicketingDate: string;
    numberOfBookableSeats: number;
    itineraries: Itineraries[];
    price: Price;
}
